### PR TITLE
[segmentation] Move towards `index_t` and `Indices`

### PR DESCRIPTION
**Pull request by mvieth**
See also commit messages. This is only one step, there is more to be done in segmentation (in another PR).

### CODE REVIEW

File: segmentation/include/pcl/segmentation/impl/grabcut_segmentation.hpp [hunk 1]
```diff
@@ -403,11 +403,11 @@ GrabCut<PointT>::computeBetaNonOrganized ()
       {
         links.nb_links = found - 1;
         links.weights.reserve (links.nb_links);
-        for (std::vector<int>::const_iterator nn_it = links.indices.begin (); nn_it != links.indices.end (); ++nn_it)
+        for (const auto& nn_it : links.indices)
```
Comment (kunaltyagi) [inline segmentation/include/pcl/segmentation/impl/grabcut_segmentation.hpp:406]: not an iterator

**Review by kunaltyagi — COMMENTED** [review]
17/20

File: segmentation/include/pcl/segmentation/impl/grabcut_segmentation.hpp (hunk 1 above)
Comment (mvieth) [reply segmentation/include/pcl/segmentation/impl/grabcut_segmentation.hpp]: Changed

File: segmentation/include/pcl/segmentation/impl/conditional_euclidean_clustering.hpp [hunk 2]
```diff
@@ -67,27 +67,27 @@ pcl::ConditionalEuclideanClustering<PointT>::segment (pcl::IndicesClusters &clus
...
+  Indices nn_indices;
   std::vector<float> nn_distances;
 
   // Create a bool vector of processed point indices, and initialize it to false
   // Need to have it contain all possible points because radius search can not return indices into indices
   std::vector<bool> processed (input_->size (), false);
 
   // Process all points indexed by indices_
-  for (int iii = 0; iii < static_cast<int> (indices_->size ()); ++iii)  // iii = input indices iterator
+  for (const auto& iii : (*indices_)) // iii = input indices iterator
```
[the remark this answers is not in the document]
Comment (mvieth) [reply segmentation/include/pcl/segmentation/impl/conditional_euclidean_clustering.hpp]: Changed

**Review by kunaltyagi — APPROVED** [review]

**Review by larshg — APPROVED** [review]

**Pull request merged**